### PR TITLE
Kernel: Handle WorkQueue OOM cases

### PR DESCRIPTION
**Pull request by supercomputer7**
Mainly a draft for now, but I will polish this a bit in the next few days.
Some places require us to handle such failure with specific hardware instructions to ensure next interactions with the hardware doesn't fail :)

cc @IdanHo

### CODE REVIEW

**Comment by IdanHo** [comment]
Nice, ref #6369

File: Kernel/Graphics/VirtIOGPU/Console.cpp [hunk 1]
```diff
@@ -65,10 +65,13 @@ void Console::enqueue_refresh_timer()
                 .width = (u32)rect.width(),
                 .height = (u32)rect.height(),
             };
-            g_io_work->queue([this, dirty_rect]() {
+            auto work_item_creation_result = g_io_work->try_queue([this, dirty_rect]() {
                 m_framebuffer_device->flush_dirty_window(dirty_rect, m_framebuffer_device->current_buffer());
                 m_dirty_rect.clear();
             });
+            if (work_item_creation_result.is_error()) {
```
Comment (bgianfo) [inline Kernel/Graphics/VirtIOGPU/Console.cpp:72]: Seems like we should assert in these cases if we don't know how to meaningfully handle these errors at this time.

Comment (supercomputer7) [reply Kernel/Graphics/VirtIOGPU/Console.cpp:72]: Right, so what do you suggest to do? keep the if statement and assert there or use MUST() wrapper?

Comment (bgianfo) [reply Kernel/Graphics/VirtIOGPU/Console.cpp:72]: I think `MUST(..)` would be fine.

**Comment by supercomputer7** [comment]
I thought about this patch and came to a conclusion I should put some effort into creating some sort of locking mechanism so in case userspace and a kernel driver try at the same time to map the ROM, this will actually load it after taking a spinlock of some sort somewhere in the Kernel.

**Pull request merged**